packages/examples/edgex/device_add_template.py

```diff
@@ -0,0 +1,189 @@
+# -*- coding: utf-8 -*-
+from liota.core.package_manager import LiotaPackage
+from random import randint
+import time
+import socket
+import logging
+import random
+import json
+import httplib
+from liota.lib.utilities.utility import systemUUID
+
+####huaqiao####
+import paho.mqtt.client as mqtt_edgex
+
+log = logging.getLogger(__name__)
+
+dependencies = ["iotcc_mqtt"]
+
+# --------------------User Configurable Retry and Delay Settings------------------------------#
+
+# The value mentioned below is the total number of Edge System being deployed in the infrastructure
+# minimum 1 for 1K, 2 for 2K, 3 for 3K, 4 for 4K and  5 for 5K Edge Systems
+no_of_edge_system_in_thousands = 1
+# Number of Retries for Connection and Registrations
+no_of_retries_for_connection = 5
+# Retry delay Min Value in seconds
+delay_retries_min = 600
+# Retry delay Max Value in seconds
+delay_retries_max = 1800
+
+# Lambda Function Multiplier uses the above settings for calculating retry and delay logic
+lfm = lambda x: x * no_of_edge_system_in_thousands
+retry_attempts = lfm(no_of_retries_for_connection)
+delay_retries = random.randint(lfm(delay_retries_min), lfm(delay_retries_max))
+
+
+#######by huaqiao zhang
+edgexmeta = {'test':'test'}
+device_name = ""
+device_value_name = ""
+device_data = 0
+
+def on_connect(client,userdata,flags,rc):
+    log.info("--------Connection from Edgex Foundry with result code" + str(rc))
+    global edgexmeta
+    client_edgex.subscribe(edgexmeta["topic"])
+
+def on_message(client,userdata,msg):
+    log.info("----------edgexfoundry msg: "+msg.topic + " " + str(msg.payload))
+    global device_name
+    global device_value_name
+    global device_data
+    edgex_data_tmp = json.loads(msg.payload)
+    if edgex_data_tmp["device"] == device_name:
+    	readings = edgex_data_tmp["readings"]
+    	readObj = readings[0]
+    	if readObj["name"] == device_value_name:
+    		device_data = readObj["value"]
+                log.info("-------- Data From EdgeX Foundry: " + str(device_data))
+
+client_edgex = mqtt_edgex.Client()
+
+def mqtt_server(user,pwd):
+	global client_edgex
+	#client_edgex = mqtt_edgex.Client()
+	client_edgex.username_pw_set(user,pwd)
+
+	client_edgex.on_connect = on_connect
+	client_edgex.on_message = on_message
+
+	client_edgex.connect("10.112.122.28",1883,60)
+	client_edgex.loop_start()
+
+
+# ---------------------------------------------------------------------------
+# This is a sample application package to publish sample device stats to
+# IoTCC using MQTT protocol as DCC Comms
+# User defined methods
+
+def device_metric():
+    global device_data
+    _t = float(device_data)
+    device_data = 0
+    return _t
+
+def http_req():
```

Comment:
Please explore if it is possible to use MQTTDeviceComms to connect between EdgeX and Liota.
Don't use http request.